### DISCUSSION TITLE
Searching by lens type not working?

### DISCUSSION THREAD

**Discussion by 1oh1** [photoprism/photoprism#2072] [Q&A] [answered]
I'm running the latest preview build of PhotoPrism (Build 220219-b52bf1bd-Linux-AMD64) and trying to [search for photos by lens type](https://docs.photoprism.app/user-guide/organize/search/#:~:text=Country%2C%20Year%2C%20Order%2C%20Camera%2C%20Lens%2C%20Color%2C%20Category)

However, in the the UI, I don't see a lens filter option and when I do a search with `lens: F2.8` I see an invalid request toast message. Was this feature removed at some point?

**Comment by graciousgrey** [comment, marked as answer]
Good catch! We should update our docs. The lens filter in the UI was replaced by a filter for the month because many users preferred it this way.
The lens search filter is looking for the lens id (it can be found in the database). So currently you can search for `lens:3`.
We could add another search filter that is looking for the lens name. Feel free to open an issue for it :)

**Reply by krakky4** [reply]
You mentioned finding the lens-id in the database. How exactly can I do this? I have photos from dozens of different lenses. the numbers do not seem to be consecutive (at least I have some numbers that do not return any results while other , higher numbers show results…) or ordered in any recreatable way (alphabet, min focal length…). So it is almost impossible for me to find photos taken with a particular lens without a proper index…
Thanks!

**Reply by graciousgrey** [reply]
You can use the lens filter to search for the lens id  `lens:2` or lens name `lens:"EF-M22mm f/2 STM"` or even partial lens names `lens:EF`

https://demo.photoprism.app/library/browse?view=cards&order=added&q=lens:2&public=true&quality=3
https://demo.photoprism.app/library/browse?view=cards&order=added&q=lens:%22EF24-105mm+f/4L+IS+USM%22&public=true&quality=3

IDs should consecutive with 1 being reserved for "Unknown". If you don't see results for some IDs this might be beause you don't have photos of this lens anymore in your library or they don't match other search filters you may have set.

**Reply by krakky4** [reply]
Thanks for the quick reply. 
Lack of consecutiveness could well be as you explained. 

However, the search by lens name or even partial lens name does not work for me. Is this a recently added feature? I am running Build 250228. 
tried with and without space, with and without quotation marks.

**Reply by graciousgrey** [reply]
The feature has been added in this release: https://docs.photoprism.app/release-notes/#may-17-2022

What is the name of the lens (you can see it on the details tab of the photo edit dialog) and what search did you try?

**Reply by krakky4** [reply]
`Lens:1` works as expected and gives me unknown lenses. 
`lens:2` gives me hundreds of photos taken with "50mm F1.7" 
However, filtering with `lens: "50mm F1.7"` (or just `lens:50` for that matter) does not return any results. 
![IMG_3049](https://github.com/user-attachments/assets/5f7aeaa6-4be7-4aa2-92be-e40a761167a3)
![IMG_3048](https://github.com/user-attachments/assets/3a5a06ad-d066-4d06-829d-12d191846144)

**Reply by graciousgrey** [reply]
Try it without the space: `lens:"50mm F1.7"`

**Reply by krakky4** [reply]
Wow - that must have been one combination I missed. Works now - thanks a lot!

For future troubleshooters: Another source of trouble for me were the „German inverted commas“ which are interpreted differently from the straight quotation marks.
